Automattic/woocommerce-payments
# Set billing details name from payerName to allow order placement in WooCommerce

Fixes #2596 

#### Changes proposed in this Pull Request

Billing details name is mandatory to complete the order in WooCommerce, but it is optional with a payment request in Stripe. Setting requestPayerName to true in the Stripe API does not set billing details name, probably due to the bug referenced here https://issueexplorer.com/issue/stripe/react-stripe-js/142

This causes order to fail when billing details name is not present in a saved payment method. The fix as suggested in the link above is to set payerName as billing details name.

#### Testing instructions
1. Sign up for Google Pay test cards or have cards without billing details name saved in Google Pay.
2. On `trunk`, check that order placement fails with the saved payment method with the error "Billing First Name is a Required Field" etc.
2. On `fix/2596-error-google-pay`, Click on "Buy with Google Pay" and complete the checkout, order should be placed without error.

-------------------


**Post merge**
